### ISSUE TITLE
Filter data and use relative abundance

### ISSUE THREAD

**Issue by wqssf102** [MGXlab/social_niche_breadth_SNB#2]
Hi,
Thank you for developing SNB algorithm，  is it possible to add the function of inputting relative abundance to calculate SNB? Because our data may be derived from metagenomic or other, and not count abundance, such as relative abundance, TPM,....
Looking forward to your help.

Thanks,
Qiusheng WU

**Comment by bastiaanvonmeijenfeldt** [comment]
Hi @wqssf102,

Thanks for your question! Currently we do not allow for this because in the paper we used an absolute abundance cut-off (of 5 reads) for the pairwise dissimilarity calculations. However, I have included the option for a relative abundance cut-off for the pairwise dissimilarity calculations in this code (with the `--c2 / --pairwise-comparisson-cutoff` set < 1) so I can implement allowance for relative abundance tables as well. I put it on the to-do list.

In the meantime, our data is also metagenomic, and the 'absolute' count refers to number of sequenced reads. So even though this is compositional data, what the script needs is read counts and total number of taxonomically annotated reads per sample in the header. This is what for example TPM is also calculated from. Could you structure your tables in that way?

As a side note, I have briefly tested a relative `--c2` cut-off and results seem to be quantitatively very similar to an absolute abundance cut-off at least for our dataset.

Best wishes,

Bastiaan

**Comment by wqssf102** [comment]
Thanks for your answer, when I use --c2 to control the relative abundance, what does the table header of the input dataset look like? Can you give a simple table as an example?

**Comment by wqssf102** [comment]
ok, i get it,thank you .

**Comment by bastiaanvonmeijenfeldt** [comment]
I have uploaded a new version of the script (v0.2) that allows for relative abundance tables as an input. The script assumes whether the input file is a count table or relative abundance table based on the header. Read count tables should have total number of taxonomically annotated reads in the header, and relative abundance tables just the sample names. It throws an error if the content does not match the header.

If a relative abundance table is supplied the `--c2 / --pairwise_comparisson_cutoff` has to be set < 1. With a read count table, `--c2 / --pairwise_comparisson_cutoff` can be set to any value. Note that in the paper, we used a cut-off of 5 reads.
